spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -81,10 +81,15 @@ def call(name)
         expect(object.foobar(:key => "value")).to equal(1)
       end
 
-      it "can accept an inner hash as a message argument" do
-        hash = {:a => {:key => "value"}}
-        expect(object).to receive(:foobar).with(:key => "value").and_return(1)
-        expect(object.foobar(hash[:a])).to equal(1)
+      if RSpec::Support::RubyFeatures.required_kw_args_supported?
+        it "can accept an inner hash as a message argument" do
+          hash = {:a => {:key => "value"}}
+          expect(object).to receive(:foobar).with(:key => "value").and_return(1)
+          # Use eval to avoid syntax error on 1.8 and 1.9
+          eval("expect(object.foobar(**hash[:a])).to equal(1)")
+          # Avoid warning: assigned but unused variable - hash
+          hash = hash
```

Comment:
If you wrap the whole spec contents in an eval don't you avoid this?

Reply:
Thanks! I've changed it as you said. https://github.com/rspec/rspec-mocks/pull/1394/commits/cdd9a0465a001f20479922ae3f6e45d1fa29f3a4